eessi/testsuite/utils.py

```diff
@@ -57,15 +57,58 @@ def is_cuda_required_module(module_name: str) -> bool:
     return requires_cuda
 
 
-def find_modules(substr: str) -> Iterator[str]:
-    """Return all modules in the current system that contain ``substr`` in their name."""
-    if not isinstance(substr, str):
+def find_modules(regex: str, name_only = True) -> Iterator[str]:
+    """
+    Return all modules matching the regular expression regex. Note that since we use re.search,
+    a module matches if the regex matches the module name at any place. I.e. the match does
+    not have to be at the start of the smodule name
+    
+    Arguments:
+    - regex: a regular expresion
+    - name_only: regular expressions will only be matched on the module name, not the version (default: True).
+    
+    Note: the name_only feature assumes anything after the last forward '/' is the version,
```

Comment:
why did you choose the last slash here?
i would think that the name is everything before the __first__ slash?

Reply:
Because I have seen things like
```
/some/prefix/name/version
```
It's not so relevant in the EESSI environment I guess, where we know it's _always_ only one slash. But for using the test suite on other module stacks, I think this gives the _biggest_ chance of doing the right thing. E.g. on Vega the native module stack has things like:

```
oneapi/advisor/latest
oneapi/advisor/2021.2.0
oneapi/advisor/2022.0.0
oneapi/advisor/2022.1.0
oneapi/mkl32/latest
oneapi/mkl32/2021.2.0
oneapi/mkl32/2021.4.0
oneapi/mkl32/2022.0.2
oneapi/mkl32/2022.1.0
```

Here, you'd _probably_ want to match things like `advisor` and `mkl`...

Again, I don't think this is a huge deal, because either choice _should_ work for EESSI, right?

Reply:
not a big deal indeed, as you can always work around it with `name_only=False`